test/tools/libtesteth/TestSuite.cpp

```diff
@@ -111,6 +111,16 @@ string const c_copierPostf = "Copier";
 
 void TestSuite::runAllTestsInFolder(string const& _testFolder) const
 {
+	// Allow to execute a custom test .json file on any test suite
+	if (!test::Options::get().singleTestFile.empty())
+	{
+		auto& testOutput = test::TestOutputHelper::get();
+		testOutput.initTest(1);
+		executeFile(fs::path(test::Options::get().singleTestFile));
+		testOutput.finishTest();
+		exit(0);
+	}
+
```

Comment:
This code should not be in `runAllTestsInFolder()`.  The caller of `runAllTestsInFolder()` already knows that the execution is not about any folder, so it can avoid calling `runAllTestsInFolder()`.

Reply:
I know. I just want it to be in one place. If you move this check outside then it will have to be added in all .cpp for all test suites.

Reply:
The move it further up, closer to `main()`.